haskell-src/Concordium/ID/AnonymityRevoker.hs

```diff
@@ -38,6 +39,31 @@ foreign import ccall unsafe "ar_info_name" arNameFFI :: Ptr ArInfo -> Ptr CSize
 foreign import ccall unsafe "ar_info_url" arUrlFFI :: Ptr ArInfo -> Ptr CSize -> IO (Ptr Word8)
 foreign import ccall unsafe "ar_info_description" arDescriptionFFI :: Ptr ArInfo -> Ptr CSize -> IO (Ptr Word8)
 foreign import ccall unsafe "ar_info_public_key" arPublicKeyFFI :: Ptr ArInfo -> Ptr CSize -> IO (Ptr Word8)
+foreign import ccall unsafe "ar_info_create"
+    createArInfoFFI ::
+        Word32 ->
+        Ptr Word8 -> CSize ->
+        Ptr Word8 -> CSize ->
+        Ptr Word8 -> CSize ->
+        Ptr Word8 -> CSize ->
+        IO (Ptr ArInfo)
+
+createArInfo :: ArIdentity -> BS.ByteString -> Text -> Text -> Text -> Maybe ArInfo
+createArInfo arId pubKey name url desc = unsafePerformIO ( do
+    let (ArIdentity idW) = arId
+    (pkPtr, pkLen) <- toByteArrayInput pubKey
+    (nPtr, nLen) <- toByteArrayInput (Text.encodeUtf8 name)
+    (urlPtr, urlLen) <- toByteArrayInput (Text.encodeUtf8 url)
+    (descPtr, descLen) <- toByteArrayInput (Text.encodeUtf8 desc)
+    ptr <- createArInfoFFI idW pkPtr pkLen nPtr nLen urlPtr urlLen descPtr descLen
+    if ptr == nullPtr
+    then return Nothing
+    else Just . ArInfo <$> newForeignPtr freeArInfo ptr)
+    where
+        toByteArrayInput bs =
+            unsafeUseAsCStringLen bs
+                $ \(p, l) -> return (castPtr p :: Ptr Word8, fromIntegral l :: CSize)
+    
```

Comment:
This won't work correctly. It might seem to work most of the time though.

The pointer you have inside `unsafeUseAsCStringLen` cannot escape the closure .So you have to nest all of these. See  how we use this elsewhere for an example.